app/Http/Controllers/AuctionController.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\AuctionRequest;
+use App\Http\Requests\UpdateAuctionRequest;
+use App\Models\auction;
+use App\Notifications\TelegramNotification;
+use App\Services\AuctionService;
+use App\Services\CategoryService;
+use App\Services\ProductService;
+use ArgumentCountError;
+use Exception;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Notification;
+use Illuminate\Validation\ValidationException;
+
+class AuctionController extends Controller
+{
+    private $auctionService;
+    private $productService;
+    private $categoryService;
+
+    public function __construct(AuctionService $auctionService, ProductService $productService, CategoryService $categoryService)
+    {
+        $this->auctionService = $auctionService;
+        $this->productService = $productService;
+        $this->categoryService = $categoryService;
+    }
+    public function index()
+    {
+        $auctions = $this->auctionService->getAll()->filter()->paginate(10);
+        // dd($auctions);
+        foreach ($auctions as $auction) {
+            $status = $auction->setStatus();
+            if ($auction->status != $status)
+                $this->auctionService->update($auction, ['status' => $status]);
+        }
+        $categories = $this->categoryService->getAll()->get();
+        return view("auctions.index", compact("auctions", "categories"));
+    }
+
+    public function create()
+    {
+        try {
+            $this->authorize("create", Auction::class);
+            $products = $this->productService->getMyProduct();
+            return view("auctions.create", compact("products"));
+        } catch (AuthorizationException $e) {
+            abort(403, 'Unauthorized');
+        }
+
+    }
+
+    public function store(AuctionRequest $request)
+    {
+        try {
+            $validated = $request->validated();
+            $product = $this->productService->getById($request->product_id);
+            $this->authorize("auctions.create", $product);
+            $this->auctionService->create($validated);
+            return redirect()->route("auctions.index")->with("success", "New auction added successfully");
+        } catch (ValidationException $e) {
+            return redirect()->back();
+        } catch (AuthorizationException $e) {
+            abort(403, 'Unauthorized');
+        }
+    }
+
+    public function show($id)
+    {
+        $auction = $this->auctionService->getById($id);
+        if ($auction)
```

Comment:
if(!$auction)
      abort(404) // or return to your custom 404 page
return view("auctions.show", compact("auction"));